docs/reference/search/search-your-data/retrievers_examples.asciidoc

```diff
@@ -0,0 +1,312 @@
+
+[source,js]
+----
+GET /retrievers_example/_search
+{
+   "retriever":{
+       "rrf": {
+           "retrievers":[
+               {
+                   "standard":{
+                       "query":{
+                           "semantic":{
+                               "field": "inference_field",
+                               "query": "Can I use generative AI to identify user intent and improve search relevance?"
+                           }
+                       }
+                   }
+               },
+               {
+                   "knn": {
+                       "field": "vector",
+                       "query_vector": [0.23, 0.67, 0.89],
+                       "k": 3,
+                       "num_candidates": 5
+                   }
+               }
+           ],
+           "rank_window_size": 10,
+           "rank_constant": 1
+       }
+   },
+   "_source": ["text", "topic"]
+}
+----
+//NOTCONSOLE
+end::basic-rrf-retriever-with-semantic-query[]
+
+tag::rrf-retriever-with-collapse[]
+[discrete]
+=== Example: Grouping results by year with `collapse`
+Let's say that we have our results, but we get back many documents for the same `year`. We can now use `collapse` with retrievers, to group results based
+on a given value and pick just the top for each sub-group!
+
+[source,js]
+----
+GET /retrievers_example/_search
+{
+   "retriever":{
+       "rrf": {
+           "retrievers":[
+               {
+                   "standard":{
+                       "query":{
+                           "semantic":{
+                               "field": "inference_field",
+                               "query": "Can I use generative AI to identify user intent and improve search relevance?"
+                           }
+                       }
+                   }
+               },
+               {
+                   "knn": {
+                       "field": "vector",
+                       "query_vector": [0.23, 0.67, 0.89],
+                       "k": 3,
+                       "num_candidates": 5
+                   }
+               }
+           ],
+           "rank_window_size": 10,
+           "rank_constant": 1
+       }
+   },
+   "collapse": {
+       "field": "year",
+       "inner_hits": {
+           "name": "topic related documents",
+           "_source": ["text", "year"]
+       }
+   },
+   "_source": ["text", "topic"]
+}
+----
+//NOTCONSOLE
+end::rrf-retriever-with-collapse[]
+
+tag::rrf-on-top-of-semantic-reranker[]
+[discrete]
+=== Example: RRF with semantic reranker
+For this scenario, let's say that we want to swap our semantic query with our `my-awesome-rerank-model` reranker that we
+have already setup. The main difference now is that, since this is a reranker, it will need an initial pool of docs to rerank!
+We know that we want to work with `ai` topics, so let's try to do just that!
```

Comment:
```suggestion

For this example, we'll replace our semantic query with the `my-awesome-rerank-model`
reranker we previously configured. Since this is a reranker, it needs an initial pool of
documents to work with. In this case, we'll filter for documents about `ai` topics.
```

Reply:
I love the enthusiasm, but exclamation marks can become visually distracting :)